library/ecp_curves.c

```diff
@@ -5548,10 +5561,22 @@ int mbedtls_ecp_mod_p448(mbedtls_mpi_uint *X, size_t X_limbs)
 
     (void) mbedtls_mpi_core_add(M, M, Q, Q_limbs);
 
-    /* Shifted carry bit from the addition is dealt with by oversize M */
+    /* Shifted carry bit from the addition fits in oversize M */
     mbedtls_mpi_core_shift_l(M, M_limbs, 224);
     (void) mbedtls_mpi_core_add(X, X, M, M_limbs);
 
+    /* Deal with carry bit by subtracting P if necessary. */
+    if (X[P448_WIDTH] != 0) {
+        mbedtls_mpi_core_sub(X, X, P, P_limbs);
```

Comment:
This is not a blocker just a comment. The functional behaviour of this method has changed, since it may or may not perform one or more reductions. This is showcased by te following modification of the test suite to remove the modular redution.

There could be value to adding a full comment in the ecp header, in line with [others ](https://github.com/Mbed-TLS/mbedtls/blob/development/library/ecp_invasive.h#L222-L238) to document that.